src/Enzo/enzo_EnzoFieldArrayFactory.cpp

```diff
@@ -0,0 +1,166 @@
+{
+  EFlt3DArray temp = full_field_from_name_(field_name);
+  if (stale_depth_ != 0){
+    return exclude_stale_cells_(temp);
+  } else {
+    return temp;
+  }
+}
+
+//----------------------------------------------------------------------
+
+EFlt3DArray EnzoFieldArrayFactory::from_grouping(Grouping &grouping,
+						 std::string group_name,
+						 int index)
+{
+  check_grouping_details_(grouping, group_name, index);
+  return from_name(grouping.item(group_name,index));
+}
+
+//----------------------------------------------------------------------
+
+EFlt3DArray EnzoFieldArrayFactory::assigned_center_from_grouping
+(Grouping &grouping, std::string group_name, int index, int dim)
+{
+  check_grouping_details_(grouping, group_name, index);
+  ASSERT("EnzoFieldArrayFactory",
+	 "reconstructed fields can only be face-centered along dim = 0, 1 or 2",
+	 dim>-1 && dim <3);
+
+  Field field = block_->data()->field();
+  std::string field_name = grouping.item(group_name,index);
+  const int id = field.field_id(field_name);
+
+  int c_arr[3]; // = {cx, cy, cz}
+  field.centering(id, c_arr, c_arr+1, c_arr+2);
+  int temp_sum = c_arr[0]+c_arr[1]+c_arr[2];
+  if ((c_arr[0] > 0) || (c_arr[1] > 0) || (c_arr[2] > 0)){
+
+    ERROR1("EnzoFieldArrayFactory::assigned_center_from_grouping",
+	   "\"%s\" has space for face-centered values on the grid exterior",
+	   field_name.c_str());
+
+  } else if (temp_sum < -1){
+
+    ERROR1("EnzoFieldArrayFactory::assigned_center_from_grouping",
+	   "the \"%s\" field is face-centered along multiple dimensions",
+	   field_name.c_str());
+
+  } else if ((temp_sum == -1) && (c_arr[dim] != -1)){
+
+    int actual_face = -1;
+    for (int i=0; i<3; i++){ if (c_arr[i] == -1) { actual_face = i; } }
+    char axis_names[3] = {'x','y','z'};
+
+    ERROR3("EnzoFieldArrayFactory::assigned_center_from_grouping",
+	   ("the underlying \"%s\" field must be cell-centered or face-"
+	    "centered along the %c-axis; not face-centered along the %c-axis"),
+	   field_name.c_str(), axis_names[dim], axis_names[actual_face]);
+  }
+
+  // separate the following if-statement from the above to avoid compiler
+  // warning about the return type
+
+  if (temp_sum == -1){
+
+    // the field's underlying centering already matches the specified centering
+    return from_name(field_name);
+
+  } else {
+
+    // the underlying field is cell-centered. Modify it's shape to match the
+    // designated centering.
+
+    int mx,my,mz;
+    field.dimensions(id,&mx,&my,&mz);
+    if (dim == 0){
+      mx--;
+    } else if (dim == 1){
+      my--;
+    } else {
+      mz--;
+    }
+
+    EFlt3DArray temp((enzo_float *) field.values(field_name), mz, my, mx);
+    if (stale_depth_ != 0){
+      return exclude_stale_cells_(temp);
+    } else {
+      return temp;
+    }
+  }
+}
+
+//----------------------------------------------------------------------
+
+EFlt3DArray EnzoFieldArrayFactory::interior_bfieldi(Grouping &grouping, int dim)
```

Comment:
I'll think about a better way to describe this

But to explain the name:
- If a cell-centered field has N elements along a given axis, then the a face-centered field must have N-1 elements along that axis or N+1 elements along that axis. The latter case includes are values on the outermost faces of the array. Basically this function returns the arrays excluding those outermost faces.
- I was doing this because the reconstructed fields never include values on those outermost faces. But maybe I can avoid this.

Reply:
Ok, to address this I made the following 2 changes:
1. I renamed the method `bfieldi_without_outermost_block_faces`
2. I added more details in the documentation of the method

I got rid of this method anyway in the new branch of the code